.github/actions/detect-workflow/Dockerfile

```diff
@@ -12,6 +12,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# NOTE: Updates to the golang base image sha256 should be verified by checking
+# the Docker official images tag details.
+# https://raw.githubusercontent.com/docker-library/repo-info/master/repos/golang/tag-details.md
```

Comment:
I'm actually not sure the best way to verify shas for official images. This was the best I could think of.

Having a  feature that parses a markdown document doesn't seem ideal.

Reply:
I was thinking of using a pre-defined policy that is "maintained" by OSSF. I think it'd help the consumer side of things. Would also help pilot the consumption side of things.

We don't need policy for everything, we can just onboard new projects as we go along.

@lumjjb @inferno-chromium wdut?

Reply:
Yeah, some kind of policy would be nice.

I'm not sure how it could be verified though since Docker doesn't seem to keep up to date with signing official images (DCT images are often very old) and there isn't any provenance or anything besides this markdown document to say that it was actually built by them and what version it is etc.

Could definitely use some more thought though. Ideally Docker would just sign or create provenance for their images with sigstore and actually keep them up to date.

Reply:
/cc @var-sdk we'd be your first user :)

Reply:
Docker with DCT used to resolve tag signatures and use only verified versions with the flag on. The longer term solution for this probably will be having [kaniko](https://github.com/GoogleContainerTools/kaniko) or docker do the same thing it did with DCT with sigstore. 

Curious about this, @tonistiigi are there any plans to have the similar functionality with sigstore? Assuming that [tag signing](https://github.com/sigstore/cosign#tag-signing) will need to be supported as well.

Reply:
In our case I'm not sure we really care too much about verifying the tag per se. I think that's more informational and matters mostly just to make sure we're using a version we support. We need to just verify the sha is ok when a bot updates it.

Security-wise, I think we really want some kind of assurance that the image at the referenced sha was built by Docker. Looking up the sha in their repo-info repository just seemed like the best option at the moment.

Though, it does seem that DCT is caught up, at least for the golang repo. Maybe they're getting better at staying up to date? So we could perhaps use it for now.
```
$ export DOCKER_CONTENT_TRUST=1

$ docker pull golang:latest
...

$ docker run golang go version
go version go1.18.4 linux/amd64

$ docker trust inspect --pretty golang:latest

Signatures for golang:latest

SIGNED TAG   DIGEST                                                             SIGNERS
latest       9349ed889adb906efa5ebc06485fe1b6a12fb265a01c9266a137bb1352565560   (Repo Admin)

Administrative keys for golang:latest

  Repository Key:       fb57d64910e2f7fa4456e938c547398305f26c15c76e9de89f76e4f32e1fd0bc
  Root Key:     c6b86f21ae272f3ae27b8da8a5762df97f09d6d0604ab49dd1d9920c6e25b65b
```

Reply:
> Curious about this, @tonistiigi are there any plans to have the similar functionality with sigstore? Assuming that [tag signing](https://github.com/sigstore/cosign#tag-signing) will need to be supported as well.

How about SLSA provenance, it a more structured and contains the tag :-)

Reply:
> In our case I'm not sure we really care too much about verifying the tag per se. I think that's more informational and matters mostly just to make sure we're using a version we support. We need to just verify the sha is ok when a bot updates it.
> 
> Security-wise, I think we really want some kind of assurance that the image at the referenced sha was built by Docker. Looking up the sha in their repo-info repository just seemed like the best option at the moment.
> 
> Though, it does seem that DCT is caught up, at least for the golang repo. Maybe they're getting better at staying up to date? So we could perhaps use it for now.
> 
> ```
> $ export DOCKER_CONTENT_TRUST=1
> 
> $ docker pull golang:latest
> ...
> 
> $ docker run golang go version
> go version go1.18.4 linux/amd64
> 
> $ docker trust inspect --pretty golang:latest
> 
> Signatures for golang:latest
> 
> SIGNED TAG   DIGEST                                                             SIGNERS
> latest       9349ed889adb906efa5ebc06485fe1b6a12fb265a01c9266a137bb1352565560   (Repo Admin)
> 
> Administrative keys for golang:latest
> 
>   Repository Key:       fb57d64910e2f7fa4456e938c547398305f26c15c76e9de89f76e4f32e1fd0bc
>   Root Key:     c6b86f21ae272f3ae27b8da8a5762df97f09d6d0604ab49dd1d9920c6e25b65b
> ```

We can try that too. If it does not work, fall back to the manual lookup.

Reply:
I decided to add a pre-submit script that verifies base images with cosign if distroless, and using DCT otherwise.